subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/AbstractBasicGroupedTaskLoggingFunctionalTest.groovy

```diff
@@ -93,7 +93,12 @@ abstract class AbstractBasicGroupedTaskLoggingFunctionalTest extends AbstractCon
         succeeds('log')
 
         then:
-        result.groupedOutput.task(':log').output == "Standard out 1\nStandard err 1\nStandard out 2\nStandard err 2"
+        if (consoleAttached && stderrAttached) {
```

Comment:
I think we're also missing some coverage that shows that the error output for two tasks running in parallel is not interleaved, but is instead grouped by task.

Reply:
I've added this coverage to the multi-project test here: https://github.com/gradle/gradle/pull/5346/files#diff-f533c2c3d1ab352e8b1f9c3fef2f87c1.  The assertions are a little complex right now, but these should simplify once we get non-console output for rich/verbose consoles grouped.